plugins/inputs/etherscan/balance.go

```diff
@@ -0,0 +1,282 @@
+package etherscan
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/big"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	ErrBalanceAddressCountExhausted = errors.New("maximum address limit reached for batch request")
+	ErrReturnCodeNotOK              = errors.New("return code was not OK")
+	ErrNotValidEthAddress           = errors.New("invalid Ethereum address")
+)
+
+type ValidatingBalanceResponder interface {
+	ValidateResponseStatus() error
+	Result() []AccountBalance
+}
+
+type balance struct {
+	Network string `toml:"network"`
+	Address string `toml:"address"`
+	Tag     string `toml:"tag"`
+}
+
+type MultiBalanceResponse struct {
+	Status  string           `json:"status"`
+	Message string           `json:"message"`
+	Results []AccountBalance `json:"result"`
+}
+
+func (mbr *MultiBalanceResponse) ValidateResponseStatus() error {
+	status, err := strconv.ParseInt(mbr.Status, 10, 64)
+	if err != nil {
+		return fmt.Errorf("error processing response status [%w]", err)
+	}
+
+	if err := validateReturnCode(status); err != nil {
+		if mbr.Message == "OK" {
+			return nil
+		}
+
+		return fmt.Errorf("%s error: [%s]", err.Error(), mbr.Message)
+	}
+
+	return nil
+}
+
+func (mbr *MultiBalanceResponse) Result() []AccountBalance {
+	return mbr.Results
+}
```

Comment:
What do you need this for? I guess it can be removed and you can access `mbr.Results` directly!?